esphome/components/ili9xxx/display.py

```diff
@@ -95,6 +96,7 @@ def _validate(config):
             cv.Optional(CONF_COLOR_PALETTE_IMAGES, default=[]): cv.ensure_list(
                 cv.file_
             ),
+            cv.Optional(CONF_DATA_RATE, default="40MHz"): spi.SPI_DATA_RATE_SCHEMA,
```

Comment:
I would strongly suggest using a safer default such as 10 MHz -- most people are connecting these with dupont wires and 40 MHz is likely to cause issues in this case. Speeds greater than 10 MHz are likely to require a custom PCB which is definitely not what most people are using. 😇

```suggestion
            cv.Optional(CONF_DATA_RATE, default="10MHz"): spi.SPI_DATA_RATE_SCHEMA,
```

Reply:
...related -- it would probably be worthwhile to add a note to the docs about this. I realize most people just want the fastest speed possible; however, there are electrical/physical limitations that software can't fix and people are likely to open issues based on this if it's not called out. 😇

Reply:
I disagree, the default was always 40MHz why should we change that to 10MHz now just because we can set the value lower.

Reply:
I didn't realize the original default was 40 MHz -- but I still think this is a risky value to use by default given how many folks who use the platform in general tend to wire projects up. Frequencies higher than about ten MHz tend to be affected negatively by the extra capacitance that breadboards/perfboards/dev boards introduce; we have seen this with the Ethernet component, for example.

I only made the suggestion because I think we should use a safer default that should work for everyone as opposed to a default that may perform better but cause problems for folks with less-than-perfect wiring...regardless of what it was going before. Still, I'm ok with the more aggressive setting, but it might be worth adding a note to the docs to try slowing it down if you experience issues with the display.

Reply:
I understand what you are saying, and i would have agreed with it when this was a new component. Sadly it is not and we should not down talk the people that are using this component to already and only want to have a faster display and not slower.

Also, in the last year or so that i have worked on the ili9xxx component and before i have never heard or seen issues that where related to the fast SPI communication.

So i belief for sure we can let it be set to 40MHz and when someone has indeed issues like you for shadow than we can now advice to set the `data rate` to a slower frequency.

Reply:
Sounds good, works for me! 😄